tests/epyccel/recognised_functions/test_math_funcs.py

```diff
@@ -0,0 +1,402 @@
+from pyccel.decorators import types
+from pyccel import epyccel
+import pytest
+from numpy.random import rand, randint
+from numpy import isclose
+
+def test_fabs_call():
+    @types('real')
+    def fabs_call(x):
+        from math import fabs
+        return fabs(x)
+
+    f1 = epyccel(fabs_call)
+    x = rand()
+    assert(isclose(f1(x) ,  fabs_call(x), rtol=1e-15, atol=1e-15))
+
+def test_fabs_phrase():
+    @types('real','real')
+    def fabs_phrase(x,y):
+        from math import fabs
+        a = fabs(x)*fabs(y)
+        return a
+
+    f2 = epyccel(fabs_phrase)
+    x = rand()
+    y = rand()
+    assert(isclose(f2(x,y) ,  fabs_phrase(x,y), rtol=1e-15, atol=1e-15))
+
+@pytest.mark.xfail
+def test_fabs_return_type():
+    @types('int')
+    def fabs_return_type(x):
+        from math import fabs
+        a = fabs(x)
+        return a
+
+    f1 = epyccel(fabs_return_type)
+    x = randint(100)
+    assert(isclose(f1(x) ,  fabs_return_type(x), rtol=1e-15, atol=1e-15))
+    assert(type(f1(x))  == type(fabs_return_type(x))) # pylint: disable=unidiomatic-typecheck
+
+def test_sqrt_call():
+    @types('real')
+    def sqrt_call(x):
+        from math import sqrt
+        return sqrt(x)
+
+    f1 = epyccel(sqrt_call)
+    x = rand()
+    assert(isclose(f1(x) ,  sqrt_call(x), rtol=1e-15, atol=1e-15))
+
+def test_sqrt_phrase():
+    @types('real','real')
+    def sqrt_phrase(x,y):
+        from math import sqrt
+        a = sqrt(x)*sqrt(y)
+        return a
+
+    f2 = epyccel(sqrt_phrase)
+    x = rand()
+    y = rand()
+    assert(isclose(f2(x,y) ,  sqrt_phrase(x,y), rtol=1e-15, atol=1e-15))
+
+
+@pytest.mark.xfail
+def test_sqrt_return_type():
+    @types('real')
+    def sqrt_return_type_real(x):
+        from math import sqrt
+        a = sqrt(x)
+        return a
+    #TODO we should use cmath instead of math
```

Comment:
Does this comment refer to both functions or just the complex function? cmath and math are 2 different packages. We need to decide if we want to support the additional package

Reply:
I see that this test fails because math does not handle complex numbers. The test therefore shouldn't exist in this file